ui/src/components/base/PreviewPopup.svelte

```diff
@@ -19,14 +21,20 @@
     }
 </script>
 
-<OverlayPanel bind:this={panel} class="image-preview" btnClose={false} popup on:show on:hide>
+<OverlayPanel bind:this={panel} class="preview {type}-preview" btnClose={false} popup on:show on:hide>
     <svelte:fragment slot="header">
         <button type="button" class="overlay-close" on:click|preventDefault={hide}>
             <i class="ri-close-line" />
         </button>
     </svelte:fragment>
 
-    <img src={url} alt="Preview {url}" />
+    {#if type === "image"}
+        <img src={url} alt="Preview {url}" />
+    {:else if type === "pdf"}
+        <object title={url.substring(url.lastIndexOf("/") + 1)} data={url} type="application/pdf">
```

Comment:
Since `url.substring(url.lastIndexOf("/") + 1)` is used below too, I think we can move it as computed/reactive declaration, something like:

```js
$: filename = url ? url.substring(url.lastIndexOf("/") + 1) : "";
```